src/main/java/hu/oe/nik/szfmv/automatedcar/math/Axis.java

```diff
@@ -0,0 +1,143 @@
+package hu.oe.nik.szfmv.automatedcar.math;
+
+import static java.lang.Math.*;
+import static java.lang.Math.PI;
+
+/**A mathematical 2-dimensional axis; either {@link Axis#X} or {@link Axis#Y}.
+ *
+ * @author Dávid Magyar - davidson996@gmail.com*/
+public enum Axis {
+
+    /**The X axis out of the two 2-dimensional axes.
+     * Usually interpreted as the horizontal axis.
+     * @see Axis#Y*/
+    X,
+
+    /**The Y axis out of the two 2-dimensional axes.
+     * Usually interpreted as the vertical axis.
+     * @see Axis#X*/
+    Y;
+
+    private final AxisVector dirPos = new AxisVector(this, true);
+    private final AxisVector dirNeg = new AxisVector(this, false);
+
+    /**Gets the unit vector of the axis towards its positive direction.*/
+    public AxisVector positiveDirection() {
+        return this.dirPos;
+    }
+
+    /**Gets the unit vector of the axis towards its negative direction.*/
+    public AxisVector negativeDirection() {
+        return this.dirNeg;
+    }
+
+    /**A vector implementation specific to the axes.
+     * @see Axis#positiveDirection()
+     * @see Axis#negativeDirection() */
+    public static final class AxisVector implements IVector {
+
+        private final Axis axis;
+        private final boolean positive;
+
+        private AxisVector(Axis axis, boolean positive) {
+            this.axis = axis;
+            this.positive = positive;
+        }
+
+        @Override public double getXDiff() {
+            return axis == Axis.X ? (positive ? +1 : -1) : 0;
```

Comment:
Tök jó, hogy egy sor, de a komplexitás ugyanaz mintha if-ekkel lenne. Ez csak egy szintaktikai cukorka. Ráadásul az egymásba ágyazott verzió szerintem már nem az a szint ami könnyű megértést támogatja.

Ráadásul a kereket fölösleges újra feltalálni. A matematikusok már definiálták az előjelfüggvényt (más néven szignum függvényt), amelynek a Java standard library-jében is van implementációja.

- https://docs.oracle.com/javase/6/docs/api/java/lang/Integer.html#signum(int)
    - ha int kell
- https://javatpoint.com/java-math-signum-method
    - ha float vagy double
    - gondolom double, mert az a visszatérési értéket, de a konstansaid int-ek igazából

Reply:
ezek mergaből jöttek, de köszi

Reply:
Megint benéztem, hogy ez a @SzFMV2020-Tavasz/team-a3 munkája.

Reply:
Az inline if-el nem komplexitást szerettem volna csökkenteni, szimplán így gyorsabb volt implementálni. Nem újra akarom feltalálni a kereket, hanem egyszerűen egyrészt nem tartom fejben, hogy melyik if-et lehetne kicserélni `jmp` helyett `invoke` utasításra, másrészt ez egyetlen `return` kifejezés és fölötte van a metódusnév ami leírja mit takar. A célom nem az előjelet visszaadni, hanem ha pozitív, akkor `+1`-et, ha negatív, akkor `-1`-et visszaadni. Bár pont ugyanez az érték amit a signum visszaad, de az eredeti szándékot a metódusnév nem írja le jól. Erőfeszítésben meg ugyanott vagyunk.